webapp/src/pages/EnableTwoFactorAuthPage/EnableTwoFactorAuthPage.tsx

```diff
@@ -0,0 +1,91 @@
+import './EnableTwoFactorAuthPage.css';
+import { useState } from 'react';
+import {
+  Button,
+  ButtonVariant,
+  Header,
+  IconVariant,
+  Input,
+  InputVariant,
+  Loading,
+} from '../../shared/components';
+import { useAuth } from '../../shared/hooks/UseAuth';
+import { TWO_FACTOR_AUTH_QR_EP_URL, USER_ME_URL } from '../../shared/urls';
+import NotFoundPage from '../NotFoundPage/NotFoundPage';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import { authApi } from '../../shared/services/ApiService';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+import { useNavigate } from 'react-router-dom';
+import { ResponseError } from '../../shared/generated';
+
+export default function EnableTwoFactorAuthPage() {
+  const { authUser, setAuthUser, isLoading } = useAuth();
+  const { goBack } = useNavigation();
+  const [code, setCode] = useState('');
+  const { warn, notify } = useNotificationContext();
+  const navigate = useNavigate();
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (
+    event,
+  ) => {
+    event.preventDefault();
+    if (!authUser) {
+      return;
+    }
+    try {
+      await authApi.authControllerEnableTwoFactorAuthentication({
+        twoFactorAuthenticationCodeDto: { code },
+      });
+      setAuthUser({ ...authUser, isTwoFactorAuthenticationEnabled: true });
+      notify('You enabled 2FA');
+      navigate(USER_ME_URL, { replace: true });
+    } catch (error: unknown) {
+      if (error instanceof ResponseError) {
```

Comment:
if we stack the changes from #444 on top of this one, I'd like to remove this block and the same one from `webapp/src/pages/ValidateTwoFactorAuthPage/ValidateTwoFactorAuthPage.tsx`
Otherwise, if you stack this one on top of that one, please, use the shared function we have there.

Reply:
Sure, if you prefer, we could merge #444 before this one (I will try to review it this afternoon) or vice versa.

Reply:
Merge this one. Mine is stacked on top of the admin page one, and thats big, so I assume it's gonna take quite longer, where instead this one is mostly ready.